js-miniapp-sdk/src/miniapp.ts

```diff
@@ -14,13 +15,36 @@ interface MiniApp {
  * A contract declaring the interaction mechanism between mini-apps and native host app to display ads.
  */
 interface Ad {
-  /** @returns The Promise of interstitial ad response result from injected side. */
-  showInterstitialAd(): Promise<InterstitialAdResponse>;
+  /**
+   * Loads the specified Interstittial Ad Unit ID.
+   * Can be called multiple times to pre-load multiple ads.
+   * Promise is resolved when successfully loaded.
+   * Promise is rejected if failed to load.
+   */
+  loadInterstitialAd<T>(id: string): Promise<T>;
```

Comment:
Is the intent to restrict the load for a specific type? Can the type be concretely defined instead?

Reply:
I tried to keep this a generic return as we don't know what will be type of return from native.

Reply:
I think that there will be nothing returned when the Ad is successfully loaded, but we should return an error type when the Ad fails to load. So maybe `Promise<null, Error>`?

Reply:
Added Promise<null | Error> as return type!